### PR TITLE
Use iszero and isone in MPFR

### PR DESCRIPTION
**Pull request by musm**
No allocations and is faster
```
julia> x = BigFloat(0)

julia> @btime iszero($x)
  9.474 ns (0 allocations: 0 bytes)
true

julia> f(x) = x == 0
f (generic function with 1 method)

julia> @btime f($x)
  99.874 ns (2 allocations: 56 bytes)
true
```

### CODE REVIEW

File: base/mpfr.jl
Comment (Sacha0) [inline base/mpfr.jl]: Perhaps nix the extra spaces and/or restore alignment with the preceding lines?

**Review by Sacha0 — APPROVED** [review]
lgtm modulo the minor comments! :)

File: base/mpfr.jl
Comment (musm) [inline base/mpfr.jl]: I have another PR that changes this, which will conflict anyways. So I prefer to see one merge first before making any additional changes.

**Comment by giordano** [comment]
I don't have anything against this patch (actually, I'm in favor of it), but on Julia 0.6 I see a fairly different benchmark result:
```julia
julia> x = BigFloat(0)
0.000000000000000000000000000000000000000000000000000000000000000000000000000000

julia> @btime iszero($x)
  9.517 ns (0 allocations: 0 bytes)
true

julia> f(x) = x == 0
f (generic function with 1 method)

julia> @btime f($x)
  9.763 ns (0 allocations: 0 bytes)
true
```
(is BenchmarkTools still broken on master?)

**Comment by musm** [comment]
@giordano My benchmarks used the commit right before BenchmarkTools broke on master

**Comment by giordano** [comment]
On https://github.com/JuliaLang/julia/commit/8d85aa47e8dc27f23daf09f0343a7868931ef9ef, on my machine:
```julia
julia> x = BigFloat(0)
0.000000000000000000000000000000000000000000000000000000000000000000000000000000

julia> @btime iszero($x)
  9.453 ns (0 allocations: 0 bytes)
true

julia> f(x) = x == 0
f (generic function with 1 method)

julia> @btime f($x)
  9.167 ns (0 allocations: 0 bytes)
true
```

**Comment by musm** [comment]
maybe we should run the BigFloat benchmarks here

**Comment by andreasnoack** [comment]
@nanosoldier `runbenchmarks("BigFloat", vs = :master)`

**Comment by andreasnoack** [comment]
@nanosoldier `runbenchmarks("BigFloat", vs=":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/23322#issuecomment-324514326) has completed - no performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/1636d2313d91d944cdd0583d359e16f739544923/5c9252e_vs_98c7a06/report.md). cc @ararslan

**Review by rfourquet — APPROVED** [review]

**Pull request merged**
